packages/core/src/common/message-service-protocol.ts

```diff
@@ -16,24 +16,37 @@ export enum MessageType {
     Log = 4
 }
 
-export const MessageClient = Symbol('MessageClient');
-export interface MessageClient {
+export interface Message {
+    type: MessageType;
+    text: string;
+    actions?: string[];
+}
+
+@injectable()
+export class MessageClient {
+
     /**
      * Show a message of the given type and possible actions to the user.
      * Resolve to a chosen action.
      * Never reject.
+     *
+     * To be implemented by an extension, e.g. by the messages extension.
      */
-    showMessage(type: MessageType, message: string, ...actions: string[]): Promise<string | undefined>;
+    showMessage(message: Message): Promise<string | undefined> {
+        // tslint:disable-next-line:no-console
```

Comment:
leftovers?

Reply:
@akosyakov I think this is the default implementation in case you haven't installed the messages extensions. Similar to QuickSelect. 
@AlexTugarev Please use the logger instead of console.